### PR TITLE
We're all one team ❤

### PR DESCRIPTION
**Pull request by KrauseFx**
**Proposed changes:**
- Already changed the GitHub teams, so that all people (Google and non-Google) have the same permissions
- No more difference between Google employed engineers and core contributors, we're all one team ❤️ 
- Have a list of all core contributors in a machine readable format to use for other things below
- List the whole team on the GitHub main README, the [docs page](https://docs.fastlane.tools) and the new [fastlane.tools website](https://fastlane.tools). Those can be auto generated based on the `.json` file
- Using that list, we could even automate adding new peeps to GitHub teams
- Automatically update them `.gemspec` to list all authors

^ All points above are proposed with this PR here, and not decided on

**Gonna do a separate PR for:**
- Grant **push access to RubyGems** to all authors, based on this list and replace the [old script](https://github.com/fastlane/fastlane/blob/master/Rakefile#L7-L14)

### CODE REVIEW

**Comment by milch** [comment]
Awesome 💯 Maybe we should also add the GitHub username, since GH and Twitter won't be equal for everyone (like me) 

> Grant push access to RubyGems to all authors, based on this list and replace the old script

I think this would be great since we won't depend on Google employees to push anymore. However, we also have to think about the chances of a compromised core contributor potentially pushing rogue updates. If we could instead have it so that CircleCI automatically pushes an update when a Version Bump commit is pushed to master, that would be even better, I think 👍 

> No more difference between Google employed engineers and core contributors, we're all one team ❤️

❤️

**Comment by lacostej** [comment]
> If we could instead have it so that CircleCI automatically pushes an update when a Version Bump
> commit is pushed to master, that would be even better, I think

Go for it!

**Comment by hjanuschka** [comment]
> No more difference between Google employed engineers and core contributors, we're all one team ❤️
:heart: super awesome.!!! best team!


i also would prefer a CI based release/push - like pushing to a tag `create_release`  or stuff like that.  in that case tests can run, and the publish procedure can run afterwards, without depending on my local machine, should reduce failure rate.

**Comment by KrauseFx** [comment]
Just verified: it's possible to add new owners using username also, so no worries re email. Seems like we want to go with the CI approach anyway, which we could do, as we already have the nightly builds set up

**Comment by KrauseFx** [comment]
The problem with deploying from a server is mostly 2 things:

1) During the `fastlane release` lane, you'll be asked for changelog, which mostly is just the commit messages, but usually we're trying to order them by importance and prettify some changes to make it easy for users to understand. Also we don't mention things that are internal only
2) We usually tweet the release (which could be automated using the `twitter` action, but we append a screenshot of the release notes)

We already have [fastlane/nightly](https://github.com/fastlane/nightly), so we already have the foundation needed to automate the release. 

I see 3 potential solutions:
1) Grant RubyGems push access to people who want to have it
2) Have the automation ready, but only use it for hotfix releases
3) Let everybody do a release, but the actual `gem push` happens from a centralized server. This requires some engineering work, but something we could do. We could use the same code as used in `nightly`, and do just the `gem push`, and the rest of the deployment happens on your local machine. This way we don't have to distribute push access to RubyGems to everybody.

**Comment by KrauseFx** [comment]
> Awesome 💯 Maybe we should also add the GitHub username, since GH and Twitter won't be equal for everyone (like me)

@milch I thought the `key` of the JSON is the GitHub user :)

**Comment by lacostej** [comment]
@googlebot I do!

**Comment by KrauseFx** [comment]
screen shot 2018-01-10 at 12 37 14 pm

File: fastlane/Fastfile
Comment (taquitos) [inline fastlane/Fastfile]: I love this 🎈:balloon::goat:

File: Rakefile
Comment (lacostej) [inline Rakefile]: I ❤️ the shuffle!

File: README.md
Comment (lacostej) [inline README.md]: where did my accents go?

Note that I only added them to see what would happen with them ;)

File: README.md
Comment (revolter) [inline README.md]: Why not use https://github.com/lacostej.png?size=140 and drop the `width` attribute?

File: Rakefile
Comment (revolter) [inline Rakefile]: Why not use https://github.com/lacostej.png?size=140 and drop the `width` attribute?

File: fastlane/Fastfile
Comment (revolter) [inline fastlane/Fastfile]: I think that the CI wants this to be `sh("bundle exec rake generate_team_table")`

File: Rakefile
Comment (milch) [inline Rakefile]: This looks really good on desktop/tablet but it's a bit small on mobile. Does anyone know enough html/css magic to make it look pretty on both? 😊

Comment (revolter) [reply Rakefile]: What exactly is small? Could you post a screenshot? As it looks ok for me.

Comment (milch) [reply Rakefile]: e.g. here the rightmost 2 columns are cut off:

screen shot 2018-01-11 at 11 37 44

Comment (milch) [reply Rakefile]: If this was iOS I'd be using a `UICollectionView` but somehow I doubt html has a `<collection-view>` tag 😄

Comment (revolter) [reply Rakefile]: It has if you want (meaning you can pun any string and it doesn't crash), but it won't have the functionality too. 😂  

I'm not sure how would this be achievable, as the HTML is obviously static and the DOM is constructed with multiple columns and I don't think you can change it with CSS. Except maybe if it uses flexbox :-?

Comment (milch) [reply Rakefile]: Yeah I did some quick testing, modifying the `generate_team_table` task like this: 

```ruby
task(:generate_team_table) do
  require 'json'
  content = ["<div id='team-table' style='display:flex; flex-wrap:wrap; flex-direction:row;' width='100%'>"]

  contributors = JSON.parse(File.read("team.json"))
  contributors.keys.shuffle.each do |github_user|
    user_content = contributors[github_user]

    content << "<div style='padding: 6px 13px; border: 1px solid #dfe2e5;'>"
    content << "<img src='https://github.com/#{github_user}.png?size=200' width=140>"
    content << "<h4 align='center'><a href='https://twitter.com/#{user_content['twitter']}'>#{user_content['name']}</a></h4>"
    # content << "<p align='center'>#{user_content['slogan']}</p>" if user_content['slogan'].to_s.length > 0

    content << "</div>"
  end
  content << "</div>"

  readme = File.read("README.md")
  readme.gsub!(%r{\<div id='team-table'.*\<\/div\>}m, content.join("\n"))
  File.write("README.md", readme)
  puts("All done")
end
```

Seems to make it work without hard-coding the row/column count, and just uses the space available

Comment (revolter) [reply Rakefile]: Something like [this][1], though it should be tested on GitHub by actually committing it, because the preview adds extra elements which breaks this.

  [1]: http://jsbin.com/zusudar/edit?html,output

Comment (janpio) [reply Rakefile]: Yep, in the "old days" we would have `float:left` on all those `div`s ;)

Comment (hjanuschka) [reply Rakefile]: does the github markdown allow div tags with inline styles?

Comment (milch) [reply Rakefile]: Ha, no, turns out you can't: https://github.com/jch/html-pipeline/blob/master/lib/html/pipeline/sanitization_filter.rb#L40

Boooooo

Comment (hjanuschka) [reply Rakefile]: have inline styles 👍 

how about using https://github.com/minimagick/minimagick and rendering a single image?, well thats drifting off a bit, don't think it should be too complicated...sorry

Comment (revolter) [reply Rakefile]: Well, yes, the best solution is: rotate your phone 90 degrees 😆

Comment (KrauseFx) [reply Rakefile]: So anything we can do, or should just leave it and ignore mobile? On docs.fastlane.tools we can have it nicer anyway, also on fastlane.tools

Comment (milch) [reply Rakefile]: Yeah not much we can do I think, since GitHub is blocking everything

File: README.md
Comment (KrauseFx) [reply README.md]: PR? :)

File: README.md
Comment (janpio) [inline README.md]: Link to Twitter homepage here.

File: README.md
Comment (janpio) [inline README.md]: README is generated? Maybe do once more so will release with current data (except if this is done on release anyway)

File: team.json
Comment (milch) [inline team.json]: Felix with his fancy slogan

**Review by janpio — APPROVED** [review]
Awesome.

Note I added 2 last comments re the Twitter links, but doesn't block merging.

File: README.md
Comment (revolter) [reply README.md]: @KrauseFx, In the JSON they exist, so the README only needs to be re-generated

File: team.json
Comment (KrauseFx) [reply team.json]: Yep, not being used yet though

File: README.md
Comment (KrauseFx) [reply README.md]: Yep, auto-generated, part of this PR

**Review by nafu — APPROVED** [review]
Love this ❤️  Best team 🚀

File: team.json
Comment (nafu) [reply team.json]: ✨

**Comment by giginet** [comment]
:heart:

**Pull request merged**
